api/src/main/java/org/openmrs/module/mirebalais/require/RequireUtil.java

```diff
@@ -66,9 +66,12 @@ public static String patientIsChild() {  // expects patient or visit to be in th
         return new String("patient.person.age < 15");
     }
 
-    // TODO: note this is really younger than or equal to...
     public static String patientYoungerThan(int age) {
-        return "patient.person.age <= " + age;
+        return "patient.person.age < " + age;
+    }
+
+    public static String patientYoungerThanMonths(int ageMonths) {
+        return "(new Date(visit.startDatetime) - new Date(patient.person.birthdate)) / (1000 * 60 * 60 * 24 * 30.4) < " + ageMonths;
```

Comment:
I do care enough about fixing this still.  If we can't use moment, that's fine, but we can build a method that is more correct.  Even if we don't think anyone will care about exact accuracy, I think we should still want it to be as accurate as possible.  What about something like this - can you run this through some tests and confirm that it is accurate for various edge case scenarios and change to it if it works?

```var d1 = new Date('2018-06-06');
var d2 = new Date('2021-05-01');

// Take the years difference and multiply by 12 months
// Then, take the months difference and add it.  This will increment or decrement from the above appropriately
// Then, if the day of the month of the future date has not reached the day of the month of the earlier date, subtract a month

var months = ((d2.getFullYear() - d1.getFullYear()) * 12) + d2.getMonth() - d1.getMonth() - (d2.getDate() < d1.getDate() ? 1 : 0);

console.log(months);```

Reply:
I just glanced at this, but it might be worth looking into loading moment into the Nashorn context. It might not even be much more work than writing the custom code (and will pay off since we can continue to use it).

https://anuradha-15.medium.com/loading-scripts-using-nashorn-85585f495cf0

(I guess we'd also have to confirm that moment is available in any basic JavaScript context where we process the rules, but I assuming it most likely will be)

Reply:
https://github.com/openmrs/openmrs-module-appframework/blob/master/api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java#L98

Reply:
Nice, that sounds good @mogoodrich . Sorry I've got to ditch this ticket though—I'm off tomorrow and offline till Tuesday. Hopefully one of you can carry it forward.

Note that the error of the naive technique never exceeds 1.4 days:

![Screenshot from 2021-09-02 18-45-53](https://user-images.githubusercontent.com/1031876/131938262-d00c46a7-a20b-4769-959f-f95a2152f338.png)

Reply:
Interesting, nice analysis...